src/shared/utilities/fileSystemWatcher.ts

```diff
@@ -0,0 +1,38 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+
+/**
+ * Interface representing an object that can listen to a vscode.FileSystemWatcher
+ */
+export interface FileWatcherListener extends vscode.Disposable {
+    onListenedChange(templatePath: vscode.Uri): Promise<void>
+    onListenedCreate(templatePath: vscode.Uri): Promise<void>
+    onListenedDelete(templtemplatePathateUri: vscode.Uri): Promise<void>
+}
+
+/**
+ * Creates a file watcher that attaches functions to onDidChange, onDidCreate, and onDidDelete functions
+ *
+ * The resulting file watcher will check the glob pattern across all workspace folders.
+ * @param listener FileWatcherListener that includes functions to attach to onEvent hooks
+ * @param globPattern Glob pattern to use for file watcher
+ */
+export function associateFileSystemWatcherWithListener(
+    watcher: vscode.FileSystemWatcher,
+    listener: FileWatcherListener,
+    globPattern: string
+): void {
+    watcher.onDidChange(async (filename) => {
```

Comment:
Calling .onXxx events from VSCode Apis return disposables that need to be handled.

In moving this method to the listener class, it might make sense to make the listener a disposable, and its dispose would appropriately dispose of the onXxx disposables.